### PR TITLE
update with rpk 24.3.1-rc1

### PR DESCRIPTION
**Pull request by Deflaimun**
## Description

Update with rpk 24.3.1-rc2

### New commands

- [rpk cluster storage cancel-mount](https://deploy-preview-819--redpanda-docs-preview.netlify.app/24.3/reference/rpk/rpk-cluster/rpk-cluster-storage-cancel-mount/)
- [rpk cluster storage list-mount](https://deploy-preview-819--redpanda-docs-preview.netlify.app/24.3/reference/rpk/rpk-cluster/rpk-cluster-storage-list-mount/)
- [rpk cluster storage mount](https://deploy-preview-819--redpanda-docs-preview.netlify.app/24.3/reference/rpk/rpk-cluster/rpk-cluster-storage-mount/)
- [rpk cluster storage status-mount](https://deploy-preview-819--redpanda-docs-preview.netlify.app/24.3/reference/rpk/rpk-cluster/rpk-cluster-storage-status-mount/)
- [rpk cluster storage unmount](https://deploy-preview-819--redpanda-docs-preview.netlify.app/24.3/reference/rpk/rpk-cluster/rpk-cluster-storage-unmount/)

### New flags
[rpk debug bundle](https://deploy-preview-819--redpanda-docs-preview.netlify.app/24.3/reference/rpk/rpk-debug/rpk-debug-bundle/
)

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-list-mount.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+- `planned`
+
+- `prepared`
+
+- `executed`
+
+- `finished`
+
+If no filter is provided, all migrations will be listed.
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-list-mount.adoc]:
```suggestion
If no filter is provided, all migrations are listed.
```

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-list-mount.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+Lists mount/unmount operations:
+[,bash]
+----
+rpk cluster storage list-mount
+----
+
+
+
+Use filter to list only migrations in a specific state:
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-list-mount.adoc]:
```suggestion
Use a filter to list only migrations in a specific state:
```

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-mount.adoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
+= rpk cluster storage mount
+
+Mount a topic to the Redpanda cluster from glossterm:tiered-storage[].
+
+This command mounts a topic in the Redpanda cluster using log segments stored in Tiered Storage. 
+
+The topic may be optionally renamed with the `--to` flag.
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-mount.adoc]:
```suggestion
You can optionally rename the topic using the `--to` flag.
```

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-mount.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Requirements:
+
+- Tiered Storage must be enabled.
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-mount.adoc]: link to https://docs.redpanda.com/current/manage/tiered-storage/#enable-tiered-storage

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-status-mount.adoc [hunk 5]
```diff
@@ -0,0 +1,45 @@
+= rpk cluster storage status-mount
+
+Status of mount/unmount operation on topic to Redpanda cluster from glossterm:tiered-storage[].
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-status-mount.adoc]:
```suggestion
Status of mount/unmount operation on topic in a Redpanda cluster from glossterm:tiered-storage[].
```

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-unmount.adoc [hunk 6]
```diff
@@ -0,0 +1,52 @@
+= rpk cluster storage unmount
+
+Unmount a topic from the Redpanda cluster and secure it in glossterm:tiered-storage[].
+
+This command performs an operation that:
+
+1. Rejects all writes to the topic.
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-unmount.adoc:7]: is order important here? If not, consider using bullets instead.

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-list-mount.adoc [hunk 7]
```diff
@@ -0,0 +1,67 @@
+= rpk cluster storage list-mount
+
+List mount/unmount operations on a topic to the Redpanda cluster from glossterm:tiered-storage[].
```
Comment (Feediver1) [inline modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-list-mount.adoc]:
```suggestion
List mount/unmount operations on a topic in the Redpanda cluster from glossterm:tiered-storage[].
```

**Review by Feediver1 — APPROVED** [review]
See suggestions

File: modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-unmount.adoc (hunk 6 above)
Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-unmount.adoc:7]: I believe it is. @kbatuigas @gene-redpanda  thoughts?

Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-cluster/rpk-cluster-storage-unmount.adoc:7]: Confirmed as the order is important. Keeping the numbered list.

**Review by gene-redpanda — APPROVED** [review]
LGTM!

**Pull request merged**